src/lib/ngsi/ContextAttributeVector.cpp

```diff
@@ -99,16 +98,20 @@ std::string ContextAttributeVector::toJsonTypes()
 
     out += JSON_STR(attrName) + ":{" + JSON_STR("types") + ":[";
 
-    std::map<std::string, int>::iterator jt;
-    unsigned int                         jx;
-    for (jt = attrTypes.begin(), jx = 0; jt != attrTypes.end(); ++jt, ++jx)
+    if (noAttrDetail == false)
     {
-      std::string type = jt->first;
-      out += JSON_STR(type);
+      std::map<std::string, int>::iterator jt;
```

Comment:
I'll suggest to avoid modifying any toJson() signature detecting the "no type condition" directly in this function. I mean, if the types vector has only one element and that element is "" (which is not legal in NGSIv2) then print `[]`.

I can try to commit the change myself if you prefer or the strategy is not clear enough...